apps/wallet/src/ui/app/staking/usePendingDelegation.tsx

```diff
@@ -0,0 +1,113 @@
+                    metadata: {
+                        fields: {
+                            name: string;
+                        };
+                    };
+                    delegation_staking_pool: {
+                        fields: {
+                            // TODO: Figure out why this is an empty string sometimes:
+                            pending_delegations:
+                                | string
+                                | {
+                                      fields: {
+                                          delegator: SuiAddress;
+                                          sui_amount: number;
+                                      };
+                                  }[];
+                        };
+                    };
+                };
+            }[];
+        };
+    };
+}
+
+interface PendingDelegation {
+    name: string;
+    staked: bigint;
+}
+
+/**
+ * Fetches the pending delegations from the system object. This is currently pretty hacky and expensive.
+ */
+export function usePendingDelegation(): [PendingDelegation[], UseQueryResult] {
+    const address = useAppSelector(({ account: { address } }) => address);
+
+    // TODO: Use generlized `useGetObject` hook when it lands:
+    const objectQuery = useQuery(['object', STATE_OBJECT], async () => {
+        return api.instance.fullNode.getObject(STATE_OBJECT);
+    });
+
+    const { data } = objectQuery;
+
+    const pendingDelegation = useMemo(() => {
+        if (
+            !address ||
+            !data ||
+            !isSuiObject(data.details) ||
+            !isSuiMoveObject(data.details.data)
+        ) {
+            return [];
+        }
+
+        const systemState = data.details.data.fields as SystemStateObject;
+
+        const validators = systemState.validators.fields.active_validators
+            .map((validator) => {
+                const pendingDelegations =
+                    validator.fields.delegation_staking_pool.fields
+                        .pending_delegations;
+
+                if (!Array.isArray(pendingDelegations)) return null;
+
+                const filteredDelegations = pendingDelegations.filter(
+                    (delegation) => delegation.fields.delegator === address
+                );
+
+                if (!filteredDelegations) return null;
+
+                // TODO: Follow-up about why this is base64 encoded:
+                const name = Buffer.from(
+                    validator.fields.metadata.fields.name,
+                    'base64'
+                ).toString();
+
+                return {
+                    name,
+                    staked: filteredDelegations.reduce(
+                        (acc, delegation) =>
+                            acc + BigInt(delegation.fields.sui_amount),
+                        0n
+                    ),
+                } as PendingDelegation;
+            })
+            .filter(notEmpty);
+
+        return validators;
```

Comment:
`validators` is a bit confusing to me - this is pending delegations per validator right?

Reply:
Good point, updated